### ISSUE TITLE
How long does it take to train yolo without pretrained model?

### ISSUE THREAD

**Issue by WEITINGLIN32** [AlexeyAB/darknet#2551]
Hello, @AlexeyAB.
I have a question and want to ask you.
The question is that I have already edited a new cfg, and I want to train it for VOC dataset without pretrained model. I also train yolov3 with the author darknet53 pretrained model, with pretrained model its mAP converges so fast. However, the model edited by me its mAP converges too slow.

The question I want to ask is that how long does it take me to train the model? And when do I need to stop training? 
Thanks a lot!

**Comment by yxwLearn** [comment]
@WEITINGLIN32 
[https://github.com/AlexeyAB/darknet#when-should-i-stop-training](url)

**Comment by yxwLearn** [comment]
@WEITINGLIN32 
[https://github.com/AlexeyAB/darknet#when-should-i-stop-training](url)

**Comment by WEITINGLIN32** [comment]
@yxwLearn 
Well, this I know. But now I am training without pretrained weight, I don't know whether stopping condition is the same or not.

**Comment by yxwLearn** [comment]
@WEITINGLIN32 
you trainning without darknet53.conv.74??

**Comment by WEITINGLIN32** [comment]
@yxwLearn 
yes, because I edit the cfg model. Therefore, I can't load the pretrained model to train my model.

**Comment by yxwLearn** [comment]
@WEITINGLIN32 
what dou you edit?

**Comment by WEITINGLIN32** [comment]
@yxwLearn I add some convolution layers.

**Comment by yxwLearn** [comment]
@WEITINGLIN32 
i'm not sure,but i think the stopping condition is the same,when you find the loss no longer decreases at many iterations ,you should stop training.

**Comment by WEITINGLIN32** [comment]
@yxwLearn 
OK, thanks your answering.
Furthermore, do you know the mAP of yolov3 training with VOC dataset?

**Comment by yxwLearn** [comment]
@WEITINGLIN32 
[https://github.com/AlexeyAB/darknet#how-to-calculate-map-on-pascalvoc-2007](url)

**Comment by WEITINGLIN32** [comment]
@yxwLearn 
I want to ask whether you know the truly mAP when yolov3 training on Pascal VOC dataset?

**Comment by AlexeyAB** [comment]
@WEITINGLIN32 

> yes, because I edit the cfg model. Therefore, I can't load the pretrained model to train my model.

If you edit only layers after 74-layer and didn't touch the first 74-layers, then you can use pre-trained weights file `darknet53.conv.74`


> The question I want to ask is that how long does it take me to train the model? And when do I need to stop training?

It depends on cfg-file (since you changed it) and your dataset.


> But now I am training without pretrained weight, I don't know whether stopping condition is the same or not.

Condition is the same - preferably to train with `-map` flag and look at the mAP graph until it stop: https://github.com/AlexeyAB/darknet/issues/url

**Comment by WEITINGLIN32** [comment]
@AlexeyAB 
- If you edit only layers after 74-layer and didn't touch the first 74-layers, then you can use pre-trained weights file darknet53.conv.74
If I edit the first 74-layers, I need to train train from scratch, right?

I also have a question, do you know how exactly the mAP when yolov3 trained with pascal VOC 2007+2012 dataset? 

THANKS @AlexeyAB

**Comment by PhilipWu526** [comment]
@WEITINGLIN32 
> If I edit the first 74-layers, I need to train from scratch, right?
In my experience, Yes.

**Comment by LEXUSAPI** [comment]
@WEITINGLIN32  @AlexeyAB  i also have the problem , is not using the pre-trained model ,it can never to get the good result in some datas . example ----voc07+12 ,i had train 10000+ it loss down ,but the map is still low .
